### PR TITLE
Create a new storage blob to track vm hashes

### PR DESCRIPTION
**Pull request by kargakis**
Create a storage blob in azure to track vm hashes since we cannot tag VMs individually.

Part of AZURE-199

/cc jim-minter asalkeld

### CODE REVIEW

**Comment by kargakis** [comment]
https://github.com/openshift/openshift-azure/issues/560

/retest

File: pkg/arm/arm.go [hunk 1]
```diff
@@ -77,10 +83,133 @@ func (g *simpleGenerator) Generate(ctx context.Context, cs *api.OpenShiftManaged
...
+			newTags := make(map[string]interface{})
+			if tags, ok := resource["tags"].(map[string]interface{}); ok {
+				for k, v := range tags {
+					newTags[k] = v
+				}
+			}
+			newTags[key] = value
+			resource["tags"] = newTags
+			added = true
+			break out
```
Comment (asalkeld) [inline pkg/arm/arm.go]: these break labels are really not intuitive, I had to go read https://golang.org/ref/spec#Break_statements to figure out it goes to the next statement out of the loop - weird.

Comment (mjudeikis) [reply pkg/arm/arm.go]: Agree :)

File: pkg/arm/arm.go [hunk 2]
```diff
@@ -77,10 +83,133 @@ func (g *simpleGenerator) Generate(ctx context.Context, cs *api.OpenShiftManaged
...
+			if !ok {
+				continue
+			}
+
+			if !isScaleSet(resource) {
+				continue
+			}
+
+			// cleanup capacity
+			deleteField(resource, "sku", "capacity")
```
Comment (mjudeikis) [inline pkg/arm/arm.go]: Just for my info, why we cleaning this one?

File: pkg/arm/arm.go (hunk 1 above)
Comment (kargakis) [reply pkg/arm/arm.go]: This can also be a return, will update

File: pkg/arm/arm.go (hunk 2 above)
Comment (kargakis) [reply pkg/arm/arm.go]: So that scaling up/down will not cause unnecessary VM rotations.

**Comment by asalkeld** [comment]
this looks fine to me, just waiting on jim's opinion.

**Comment by kargakis** [comment]
/hold

After some testing, this approach seems useless. When a tag gets applied in a scale set, the underlying VMs get their tags updated automatically. :/

**Comment by asalkeld** [comment]
> /hold
> 
> After some testing, this approach seems useless. When a tag gets applied in a scale set, the underlying VMs get their tags updated automatically. :/

That sucks :-(

**Comment by kargakis** [comment]
Keeping the tag in the scale set is probably something that we still need to do. We need to use it when comparing with the VM tags, just need to track VM tags elsewhere and not on the VMs.

/hold cancel

**Comment by kargakis** [comment]
Updated to add the scale set hash directly in the storage blob as discussed with @jim-minter, should be ready for a review

**Comment by jim-minter** [comment]
Given that #683 builds on this, and I've done a detailed review there, I would be inclined to close this in favour of getting #683 through in one go.

/hold

**Pull request closed without merging**